### ISSUE TITLE
`null` String for `Exception`s deserialized as String "null" instead of `null`

### ISSUE THREAD

**Issue by ZeleniJure** [FasterXML/jackson-databind#1842]
Hi!
I have been trying to solve this problem for over a day now, and what research I did showed that this should work, but it does not in my case, e.g. default behaviour should deserialize null objects to null objects, right?

The problem is that while serialization of string object properties with null value is fine, the deserialization is not.

What I want is, if string property value

- is "null" -> deserialize to string "null"
- is null -> deserialize to null object
- is "" -> deserialize to empty string ("")
- is "ladida" -> deserialize to string "ladida"

Example code to reproduce the error:
```
        final ObjectMapper mapper = new ObjectMapper();
        // so that the error is visible at the beginning of the output :)
        mapper.configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true);
        final NullPointerException ex = new NullPointerException();
        String json;
        try {
            json = mapper.writeValueAsString(ex);
            System.out.println("OK\t-> " + json);
            final NullPointerException parsed = mapper.readValue(json, NullPointerException.class);
            json = mapper.writeValueAsString(parsed);
            System.out.println("Not OK\t<- " + json);
        } catch (final IOException e1) {
            e1.printStackTrace();
        }
```

Output:
See 'localizedMessage' and 'message' strings in serialized and deserialized output. Serialization is OK (null), deserialized is not ok (string containing "null")
```
OK	-> {"cause":null,"localizedMessage":null,"message":null,"stackTrace":[{"className":"com.cosylab.tcs.commons.communication.model.Test","fileName":"Test.java","lineNumber":87,"methodName":"main","nativeMethod":false}],"suppressed":[]}
Not OK	<- {"cause":null,"localizedMessage":"null","message":"null","stackTrace":[{"className":"com.cosylab.tcs.commons.communication.model.Test","fileName":"Test.java","lineNumber":87,"methodName":"main","nativeMethod":false}],"suppressed":[]}
```

Version: 2.9.1

**Comment by cowtowncoder** [comment]
Jackson json module does not coerce JSON `null`s into String `"null"` so I am bit surprised by this.

But I think it may be possible that specific deserializer for `Exception` types might have something, so there may be this one special case. I'll have a look if that explains what you observe.

**Comment by cowtowncoder** [comment]
Ok yes. Use of `parser.getText()` unfortunately did include `"null"` as value. So better choice is `parser.getValueAsString()` which only coerces scalars, but returns `null` for JSON `null` as well as for structured tokens (START_ARRAY, START_OBJECT).

**Comment by sanjaybsm** [comment]
is this issue fixed, if yes then could you please let us know the version?

**Comment by kupci** [comment]
According to the doc it was fixed in 2.9.3: https://github.com/FasterXML/jackson-databind/blob/master/release-notes/VERSION-2.x

**Comment by cowtowncoder** [comment]
@sanjaybsm see the "Milestone" on right side of this page. That is where version information is usually included. Fix is also listed under `release-notes/VERSION` (or depending on branch, `release-notes/VERSION-2.x`).
